### PR TITLE
BP-70: Directly use `(de)serialize` on ontology types

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

As a quick solution to distinguish between format validation and content validation, the raw module should be made private and the serde implementation should use that internally. This greatly improves the DX around the types.
The amount of code which can be removed because of this speaks for itself.

## 🔍 What does this change?

- Avoid the steady back and forth casting between `raw::` and normal representation

## 🚫 Blocked by

- https://github.com/blockprotocol/blockprotocol/pull/1350

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: apps/hash-graph/lib/type-fetcher/src/fetcher.rs [hunk 1]
```diff
@@ -29,9 +29,9 @@ impl fmt::Display for FetcherError {
 #[derive(Debug, Serialize, Deserialize)]
 #[serde(untagged)]
 pub enum OntologyTypeRepr {
```
Comment (indietyp) [inline apps/hash-graph/lib/type-fetcher/src/fetcher.rs:31]: can probably be renamed, as it is no longer a `repr`, but just a plain `OntologyType`

**Review by indietyp — CHANGES_REQUESTED** [review]
just a minor comment

File: apps/hash-graph/lib/type-fetcher/src/fetcher.rs (hunk 1 above)
Comment (TimDiekmann) [reply apps/hash-graph/lib/type-fetcher/src/fetcher.rs:31]: `OntologyType` is already the name of a trait so I left this for now.

Comment (TimDiekmann) [reply apps/hash-graph/lib/type-fetcher/src/fetcher.rs]: I renamed it to `FetchedOntologyType`. It's only used in the type fetcher and it's better than having `Rear` as part of the name. We can always rename it or refactor it if we come up with another idea 🙂

**Review by indietyp — APPROVED** [review]

**Pull request merged**